tests/Http/User/UserIdentityControllerTest.php

```diff
@@ -0,0 +1,145 @@
+use ValueObjects\StringLiteral\StringLiteral;
+use ValueObjects\Web\EmailAddress;
+use Zend\Diactoros\Response\JsonResponse;
+use Zend\Diactoros\ServerRequest;
+
+class UserIdentityControllerTest extends TestCase
+{
+    /**
+     * @var UserIdentityController
+     */
+    private $userIdentityController;
+
+    /**
+     * @var UserIdentityResolver
+     */
+    private $userIdentityResolver;
+
+    protected function setUp(): void
+    {
+        $this->userIdentityResolver = $this->createMock(UserIdentityResolver::class);
+
+        $this->userIdentityController = new UserIdentityController(
+            $this->userIdentityResolver,
+            'current_user_id'
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_can_get_user_identity_by_email(): void
+    {
+        $userIdentity = new UserIdentityDetails(
+            new StringLiteral('user_id'),
+            new StringLiteral('jane_doe'),
+            new EmailAddress('jane.doe@anonymous.com')
+        );
+
+        $this->userIdentityResolver->expects($this->once())
+            ->method('getUserByEmail')
+            ->willReturn($userIdentity);
+
+        $response = $this->userIdentityController->getByEmailAddress(
+            (new ServerRequest())->withAttribute('emailAddress', 'jane.doe@anonymous.com')
+        );
+
+        $this->assertJsonResponse(new JsonLdResponse($userIdentity), $response);
+    }
+
+    /**
+     * @test
+     */
+    public function it_returns_not_found_on_get_by_email_when_email_is_missing(): void
+    {
+        $response = $this->userIdentityController->getByEmailAddress(new ServerRequest());
+
+        $this->assertJsonResponse($this->createUserNotFoundProblem(), $response);
+    }
+
+    /**
+     * @test
+     */
+    public function it_returns_not_found_on_get_by_email_when_user_identity_not_found(): void
+    {
+        $this->userIdentityResolver->expects($this->once())
+            ->method('getUserByEmail')
+            ->with(new EmailAddress('jane.doe@anonymous.com'))
+            ->willReturn(null);
+
+        $response = $this->userIdentityController->getByEmailAddress(
+            (new ServerRequest())->withAttribute('emailAddress', 'jane.doe@anonymous.com')
+        );
+
+        $this->assertJsonResponse($this->createUserNotFoundProblem(), $response);
+    }
+
+    /**
+     * @test
+     */
+    public function it_can_get_user_identity_of_current_user(): void
+    {
+        $userIdentity = new UserIdentityDetails(
+            new StringLiteral('current_user_id'),
+            new StringLiteral('jane_doe'),
+            new EmailAddress('jane.doe@anonymous.com')
+        );
+
+        $this->userIdentityResolver->expects($this->once())
+            ->method('getUserById')
+            ->with(new StringLiteral('current_user_id'))
+            ->willReturn($userIdentity);
+
+        $response = $this->userIdentityController->getCurrentUser();
+
+        $userIdentityAsArray = $userIdentity->jsonSerialize();
```

Comment:
Nit: There's also no test for `UserIdentityDetails`, so this can return whatever. Maybe hardcode the expected JSON (as an array) instead, so we don't get a self-fulfilling prophecy here.

(For example I could make `jsonSerialize()` return an empty array and nowhere would that cause a test to fail)

We could also add tests specifically for `UserIdentityDetails` but I think it's just fine to hardcode the expected JSON here, it's also easier to read at a glance.

Reply:
Good catch! I added a method to convert to an array and also used that inside the `it_can_get_user_identity_by_email` test.